sui/tests/shared_objects_tests.rs

```diff
@@ -419,3 +425,77 @@ async fn shared_object_sync() {
         }
     }
 }
+
+#[tokio::test]
+async fn shared_object_on_gateway() {
+    let mut gas_objects = test_gas_objects();
+
+    // Get the authority configs and spawn them. Note that it is important to not drop
+    // the handles (or the authorities will stop).
+    let (configs, key_pairs) = test_authority_configs();
+    let _handles = spawn_test_authorities(gas_objects.clone(), &configs, &key_pairs).await;
+    let clients = create_authority_clients(&configs);
+    let path = tempfile::tempdir().unwrap().into_path();
+    let gateway = Arc::new(GatewayState::new_with_authorities(path, clients).unwrap());
+
+    // Publish the move package to all authorities and get the new package ref.
+    tokio::task::yield_now().await;
+    let package_ref = publish_counter_package(gas_objects.pop().unwrap(), &configs).await;
+
+    // Send a transaction to create a counter, but only to one authority.
```

Comment:
Do we send this transaction to a single authority? If so, which one?

Reply:
It sends to the gateway, which then sends to a quorum of authorities.